### ISSUE TITLE
TBD public event calendar integration

### ISSUE THREAD

**Issue by dayhaysoos** [TBD54566975/developer.tbd.website#340]
# Summary

There currently is no way for the open source community to find out about "other" events, such as office hours and other public meetings that may be interesting.

## Solution

The TBD developer website should have a place to surface our public calendar. This would allow our users to have insight on when and where we do things and give them an opportunity to join us.

## Proposal
We integrate the TBD public calendar by pulling the data and displaying it on the website. 

The data will soon come from a service we're standing up; it's a small API to handle fetching of events from the Google Calendar API. Requests from the dev site will hit that service and render the events on a new page. This issue is to create that page.

Until the service is up and running, we may build the front end page by using the following mock data which contains 5 events:

```
[{"start":"2023-10-13T09:00:00-07:00","end":"2023-10-13T09:45:00-07:00","summary":"Hackfest Fridays: Q&A + PR Reviews","hangoutLink":"https://meet.google.com/tez-jjjq-kea","htmlLink":"https://www.google.com/calendar/event?eid=MGU0Yjd2M2Y5OGU0dnZhdGFja2xtbGNnaGggY19iMmI4MzY3ZGFjNTM2ZWJmNzU3YjI3NDVmY2M1ZmJmZjIwOTlmNmM1NzRiYzEzZjgzZDE2NDIzZGIyZGQ1NTM1QGc","location":"VC: tbd-stage","description":"Are you interested in making a contribution during Hacktoberfest? We are here to help you with: \n 🌐 Version Control \n📝 Pull Requests \n🧹 Commits \n🐞 Debugging\n ✨  Any questions that you have"},{"start":"2023-10-13T11:00:00-07:00","end":"2023-10-13T11:30:00-07:00","summary":"Weekly Delivery - Open Source Programs","htmlLink":"https://www.google.com/calendar/event?eid=NzhuZzhqcHBjY2U1NXRkZXA1YWQ5bnU0Z29fMjAyMzEwMTNUMTgwMDAwWiBjX2IyYjgzNjdkYWM1MzZlYmY3NTdiMjc0NWZjYzVmYmZmMjA5OWY2YzU3NGJjMTNmODNkMTY0MjNkYjJkZDU1MzVAZw","location":"https://discord.com/channels/937858703112155166/1008794207462359170"},{"start":"2023-10-17T10:30:00-07:00","end":"2023-10-17T11:00:00-07:00","summary":"Weekly Scoping - Open Source Programs","htmlLink":"https://www.google.com/calendar/event?eid=N2ZkOXVwbnBxaHFia2EzcXFsdXU1cWx2MThfMjAyMzEwMTdUMTczMDAwWiBjX2IyYjgzNjdkYWM1MzZlYmY3NTdiMjc0NWZjYzVmYmZmMjA5OWY2YzU3NGJjMTNmODNkMTY0MjNkYjJkZDU1MzVAZw","location":"https://discord.com/channels/937858703112155166/1008794207462359170"},{"start":"2023-10-17T11:00:00-07:00","end":"2023-10-17T11:45:00-07:00","summary":"DWN Office Hours","htmlLink":"https://www.google.com/calendar/event?eid=NGVidGZwaG0waWhzY2cybG5lYmF0OGg2ZTlfMjAyMzEwMTdUMTgwMDAwWiBjX2IyYjgzNjdkYWM1MzZlYmY3NTdiMjc0NWZjYzVmYmZmMjA5OWY2YzU3NGJjMTNmODNkMTY0MjNkYjJkZDU1MzVAZw","location":"https://discord.com/channels/937858703112155166/937858703820980297","description":"In Discord #tbdiscussions"},{"start":"2023-10-19T09:00:00-07:00","end":"2023-10-19T10:00:00-07:00","summary":"Open Protocols Workshop: Unlocking Music Sharing","htmlLink":"https://www.google.com/calendar/event?eid=MjYwdXB0Y29obm9qOWJ0Mm9sZzVjZ2dqZXMgY19iMmI4MzY3ZGFjNTM2ZWJmNzU3YjI3NDVmY2M1ZmJmZjIwOTlmNmM1NzRiYzEzZjgzZDE2NDIzZGIyZGQ1NTM1QGc","location":"https://us02web.zoom.us/webinar/register/1916951491455/WN_Ta8RwGqQSA6iEGe_v19v1w#/registration","description":"<p>We're partnering with Women Who Code to host Open Protocols Workshop: Unlocking Music Sharing. Join our speakers across Square, TBD, and TIDAL on October 19th as they lead a discussion on how to build a playlist app using Web5.js.</p><p><br></p>"}]
```

**Comment by ALRubinger** [comment]
@dayhaysoos estimates 2 weeks for the above. There's a potential Phase 1 we can do for this that embeds the calendar:

* Create separate section under "Events" for "Meetings" (or other name)
* Embed Google Calendar's widget there
* Put a heading on it
* ...and a link for the community to subscribe to the Google Calendar: https://calendar.google.com/calendar/u/0?cid=Y19iMmI4MzY3ZGFjNTM2ZWJmNzU3YjI3NDVmY2M1ZmJmZjIwOTlmNmM1NzRiYzEzZjgzZDE2NDIzZGIyZGQ1NTM1QGdyb3VwLmNhbGVuZGFyLmdvb2dsZS5jb20

It's ugly but its gets the community the info they need very quickly.

Discuss w/ @angiejones

**Comment by angiejones** [comment]
pull me in next time this is discussed. wanna reconcile this with the purpose of the current Events page

**Comment by dayhaysoos** [comment]
Had a great call with @angiejones to get more clarity on what is valuable to her as far as the events pages go.

Takeaways:
- Would be nice to have something a little more automated, as of right now she's the only one that puts events on the docs site.
    - Automation might look like pulling from the Public events calendar when anyone from TBD updates it
- Need to be able to differentiate between types of events. Is it a one off event for a conference or a repeating event for a set amount of time? (Probably need to handle this with "occurrence" labels from Google Calendar.


Also, had to open up a support ticket to Corp Sec to figure out the best way to go about getting Google Calendar API keys. Still waiting for a response.

**Comment by dayhaysoos** [comment]
CorpSec asked some questions about what it's for. They replied with saying they contacted the CAE team. Waiting for the update now.


In the mean time, I will be spinning up my own instance just to get everything set in place so that all we'll need is those API keys when the time comes.

**Comment by dayhaysoos** [comment]
Was given permission to create API key, however, I seem to be locked out of other systems required to do so. Still blocked on this but will be working with the appropriate teams to resolve.

**Comment by dayhaysoos** [comment]
Spoke to @angiejones about this today. This events feature is important but we don't want it to get too much in the way of the work that needs to be done for the docs section, there are also tons of hurtles I'm going through like getting the keys from CorpSec.

We talked about possibly figuring out a way to add to the calendar feature in discord to get something out the door sooner than later, going to investigate that

**Comment by EbonyLouis** [comment]
our public OS calendar events are currently linked with our discord calendar so that part is done

**Comment by ALRubinger** [comment]
Outdated
